server/repository/src/db_diesel/changelog/README.md

```diff
@@ -0,0 +1,43 @@
+# Changelog
+
+A number of operations rely on processing a `queue` of database changes.
+These operations include pushing records to central server during synchronisation, processing transfer records, processing messaging, aggregation, interoperability synchronisation, etc.. 
+
+Generic mechanism for recording and querying these database changes was implemented in the form of the `changelog` table.
+
+## Triggers
+
+Database triggers insert new entries into changelog table with:
+
+* cursor <- sequential number for order of database operation
+* record_id 
+* table_name
+* store_id
+* name_id
+
+## changelog_deduped view
+
+Current use cases require only the latest operation for a particular record, thus a database view is used to query latest operation grouped by unique record_id.
+
+`deduping` is short for deduplication (process of removing identical entries)
+
+## name_id and store_id
+
+Some consumers of changelog need to filter database operations based on `ownership` of the record on current site. 
+* Synchroniser should only push records belonging to current site
+* Transfer should only process records that are destined for current site
+
+name_id and store_id is stored against records that need to be filtered in this fashion against database operation in changelog. Extra logic is added to triggers for these records to transfer store_id and name_id to changelog.
+
+For some data tables it's not possible to have foreign records on the site, thus name_id and store_id are optional and are set to null for those cases.
+
+## Cursors
```

Comment:
If you are targeting understanding for someone completely new to the code base, I would be tempted to move the Cursors section up earlier in the document.
And I'd add a quick explanation something like this:

```
The 'cursor' is used by changelog consumers to track which records they have already processed and allow them to start processing again from where they left off.
For example, if the requisition processor has processed all the records until cursor 10, it should record the last cursor it processed (e.g. 10). Next time it starts processing records from the changelog, it will start from cursor number 11. 
Meanwhile the shipment processor could be further ahead, processing cursor number 50. By allowing the processors to track their own cursor value, we don't need a separate queue of unprocessed records for each processor.
 ```

Reply:
Thanks: 5707429b6383efb1ddbd03b77b08316453bfc2fe

> If you are targeting understanding for someone completely new to the code base

Hmm, this is more of an advanced topic in my opinion, but yeah good to be more specific.